### ISSUE TITLE
🐛[BUG][Combat] Extremely inconsistent Aspidochelone behavior

### ISSUE THREAD

**Issue by cairthenn** [HorizonFFXI/HorizonXI-Issues#1833]
**What does this bug affect?**

Aspidochelone currently seems to have two modes:

- Mode 1: Aspidochelone will enter its shell after a damage threshold (3k?) or timer (60 seconds?) has been reached. It will exit its shell on the same condition.
- Mode 2: Aspidochelone will never enter its shell on its own. After a damage threshold is reached, it will enter and immediately exit its shell simultaneously.

The pacing for the fight is very different depending on how it is behaving, and there doesn't seem to be a clear trigger to push it from one mode to the other.

It seems likely to me that "mode 1" is the correct Aspidochelone behavior, and then something happens to cause it be stuck in "mode 2", which is actually just a bugged state.

**Steps to Reproduce**
- Not clear. Behavior changes inconsistently throughout the fight.
- For example, on todays kill, it was in mode 2 until 30%, then it got stuck in mode 1 and regenned to over 50%,. After some damage attempts, it got stuck in mode 2 again until it died.

**Expected behavior**
If Aspidochelone script is intended to mirror the old script/KS99 script, it should behave as per "mode 1" for the entire fight.

**Comment by cairthenn** [comment]
So, after the discussion, I'd imagine the script is mostly working as intended. The point of contention (and confusion), I suppose, is when damage counts towards each threshold.

It seems there is no grace period on enter/leave, so a burst in its shell that does almost double the necessary damage will leave it prime to enter its shell again after a few melee swings.

Similarly, a turtle that has freshly exited its shell can take ~4k from a SC/MB and then immediately exit after its entered.

If this is the intended behavior, it is likely not "bugged" and simply requires groups to be extremely precise about how many people are bursting when.

**Comment by cairthenn** [comment]
My outstanding concern here is, despite the script probably working as intended to be written, is the fight acting in the way that is intended?

Currently, the strength of BLM tends toward two scenarios:
- If you burst in while in the shell, you are likely going to push it out of the shell *and almost back in, to the point where a few melee swings will cause a phase shift*.
- Instead, if you simply burst when it's not in the shell, you can do nearly 4k damage in a magic burst and skip the shell phase entirely.

Basically, a successful kill with the current design of the fight should never see a group planning for the mob to enter its shell, but simply doing enough damage to ignore it.

One potential "fix" would involve a slight grace period after entering or exiting, where damage is not immediately counted toward the new threshold.

If the fight is acting as intended, and people are expected to monitor overflow damage on the shell in/out and only do extremely controlled amounts of damage, then this can likely be closed.
